### ISSUE TITLE
"aria-required-children" failure on role "listbox" for missing child nodes in the DOM 

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#874]
```aria-required-children``` fails for a ```listbox``` not having a child role ```option```. 
In a few cases authors tend to create the child ```option```elements on the fly. 

So, on roles like ```listbox``` should we tweak ```aria-required-children``` rule to:
- Needs Review: If a ```listbox``` has no child elements that are exposed to the ATs
- Fail: If a ```listbox``` has child elements exposed to ATs and is missing role ```option```

Link to live demo: https://www.bing.com/

<pre><code>axe-core version: 3.0.0-beta.3
axe-Coconut version: 3.3.0-beta.0
</code></pre>

**Comment by marcysutton** [comment]
We talked about making the no child elements thing a Review Item, instead of passing it. That would prompt developers to test again and see if it's really a failure.

**Comment by iamrafan** [comment]
My bad, I edited it to needs review.

**Comment by dylanb** [comment]
Wouldn't use of aria-busy be the appropriate way to handle this https://www.w3.org/TR/wai-aria-1.1/#aria-busy ?

**Comment by liloda** [comment]
Sounds like a good idea, but adding aria-busy="true" to the parent (with role="listbox") didn't prevent it failing aria-required-children. So I guess the rule would need to be updated.

**Comment by dylanb** [comment]
@liloda I am proposing that we add support for that as a solution for this problem

**Comment by WilcoFiers** [comment]
Agreed. An extra check that allows this to pass if its inside an aria-busy="true" would make sense.

**Comment by iamrafan** [comment]
I don't think targeting ```aria-busy``` would work, as it is not a required attribute on role ```listbox```. Cases like [Single-Select Listbox]( https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html#ex1_label) need not use it.

**Comment by WilcoFiers** [comment]
It shouldn't be required. What I think Dylan is suggesting is that axe-core only allow elements not to have their required children if they have `aria-busy` on them (or are contained within an element with `aria-busy`). Even with `aria-busy` on it I would put it for review, and not outright pass it so you couldn't just wrap a page in `aria-busy` to get everything to pass.

**Comment by liloda** [comment]
I think we could convince teams to use aria-busy if it meant they didn't fail. The challenge will be communicating this method to users who fail aria-required-children.

**Comment by dylanb** [comment]
@marcysutton the empty listbox role is different from this because an empty listbox is a valid final state for that UI control.

@iamrafan the reason I am suggesting adopting this is because this is precisely what `aria-busy` was intended for i.e. for the code to indicate when the state of the UI control is not yet complete but is expected to become complete at some future point.

**Comment by marcysutton** [comment]
@dylanb how do you differentiate the two if we're allowing empty listboxes?

**Comment by WilcoFiers** [comment]
I agree with Dylan that listbox should be allowed empty. That's the change I'll make for this. I'm going to raise a separate issue for aria-busy, because it has implications for many more rules.

**Comment by iamrafan** [comment]
@WilcoFiers @straker : This issue seems to have regressed. ```aria-required-children``` is failing for an empty ```listbox``` on https://www.bing.com/ in axe-core 3.3.1, instead of being shown as a needs review item. Please look into this issue.

![image](https://user-images.githubusercontent.com/22068434/61969513-923a7700-af8f-11e9-9d4d-8dbba4dfd241.png)
